app/src/main/java/org/glucosio/android/activity/RemindersActivity.java

```diff
@@ -0,0 +1,135 @@
+                        cal.get(Calendar.MINUTE),
+                        is24HourFormat);
+                tpd.show(getFragmentManager(), "Timepickerdialog");
+            }
+        });
+
+        listView = (ListView) findViewById(R.id.activity_reminders_listview);
+        listView.setEmptyView(findViewById(R.id.activity_reminders_listview_empty));
+        listView.setAdapter(presenter.getAdapter());
+        addFab.postDelayed(new Runnable() {
+            @Override
+            public void run() {
+                AnimationTools.startCircularReveal(addFab);
+            }
+        }, 600);
+    }
+
+    public void updateReminder(Reminder reminder) {
+        presenter.updateReminder(reminder);
+    }
+
+    public void updateRemindersList() {
+        listView.setAdapter(presenter.getAdapter());
+        listView.invalidate();
+    }
+
+    @Override
+    public void onTimeSet(RadialPickerLayout view, int hourOfDay, int minute, int seconds) {
+        Calendar cal = Calendar.getInstance();
+        cal.set(Calendar.HOUR_OF_DAY, hourOfDay);
+        cal.set(Calendar.MINUTE, minute);
+        // Id is HOURS+MINUTES to avoid duplicates
+        String concatenatedId = hourOfDay + "" + minute;
+        // Metric is always glucose until I write support for other metrics so...
+        // TODO: Add Reminders for other metrics
+        // Also oneTime is always set to false until I implement one time alarms
+        // TODO: Implement one time alarms
+        presenter.addReminder(Long.parseLong(concatenatedId), cal.getTime(), "glucose", false, true);
+    }
+
+    public void showDuplicateError() {
+        View parentLayout = findViewById(R.id.activity_reminders_root_view);
+        Snackbar.make(parentLayout, "This reminder already exists.", Snackbar.LENGTH_SHORT).show();
+    }
+
+    public void showBottomSheetDialog(final long id) {
+        final BottomSheetDialog mBottomSheetDialog = new BottomSheetDialog(this);
+        View sheetView = getLayoutInflater().inflate(R.layout.fragment_reminders_bottom_sheet, null);
+        LinearLayout delete = (LinearLayout) sheetView.findViewById(R.id.fragment_history_bottom_sheet_delete);
+        delete.setOnClickListener(new View.OnClickListener() {
+            @Override
+            public void onClick(View view) {
+                presenter.deleteReminder(id);
+                updateRemindersList();
+                mBottomSheetDialog.dismiss();
+            }
+        });
+
+        mBottomSheetDialog.setContentView(sheetView);
+        mBottomSheetDialog.show();
+    }
+
+    @Override
+    public boolean onCreateOptionsMenu(Menu menu) {
+        getMenuInflater().inflate(R.menu.menu_reminders, menu);
+        return true;
+    }
+
+    @Override
+    public boolean onOptionsItemSelected(MenuItem menuItem) {
+        if (menuItem.getItemId() == android.R.id.home) {
+            finish();
+        } else if (menuItem.getItemId() == R.id.action_menu_save){
+            presenter.saveReminders();
```

Comment:
That is not clear, why do we have "Save" is not adding alarm item is enough? Actually, it is saved in Realm but not scheduled in Alarm manager?